src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs

```diff
@@ -1778,16 +1778,9 @@ private IEnumerable<TypeInferenceInfo> InferTypeInYieldStatement(YieldStatementS
 
                 var memberType = GetMemberType(memberSymbol, out _);
 
```

Comment:
```suggestion
                // We don't care what the type is, as long as it has 1 type argument. This will work for IEnumerable, IEnumerator,
                // IAsyncEnumerable, IAsyncEnumerator and it's also good for error recovery in case there is a missing using.
```